src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
I'm not a big fan of this "ID" thing regarding the input devices...

I'd rather follow the "seat" route of libinput (which matches pretty closely the wayland protocol, probably by design): there can be one or more logical `Seat`s, but each of them has at most 1 of each input device type.

In this case, we could rather have a `SeatHandler` trait, with a type of which an instance is created for each seat, so the trait would probably have to include a constructor method.

In the glutin case, there would be only one seat.